src/pudl/output/pudltabl.py

```diff
@@ -129,6 +131,35 @@ def __init__(
         # Used to persist the output tables. Returns None if they don't exist.
         self._dfs = defaultdict(lambda: None)
 
+        self._register_output_methods()
+
+    def _register_output_methods(self):
+        """Load denorm assets and register methods for retrieving each asset."""
+        denorm_assets = {
+            "ferc1": load_assets_from_modules(
+                [pudl.output.denorm_ferc1], group_name="ferc1"
+            ),
+            "eia": load_assets_from_modules([pudl.output.denorm_eia], group_name="eia"),
+        }
+
+        for group, assets in denorm_assets.items():
+            for key in pudl.helpers.get_asset_keys(assets):
+                asset_name = key.path[0]
```

Comment:
Instead of indexing the path attribute, we can use `key.to_python_identifier()` which just merges the elements in path together.

Reply:
Thanks, this is much better!